src/GraphNodeManagerViewExtension/ViewModels/GridNodeViewModel.cs

```diff
@@ -318,8 +404,19 @@ private bool IsNodeOutputFunction(MirrorData mirrorData)
         {
             if (mirrorData == null || mirrorData.Class == null) return false;
 
-            if (string.Equals(mirrorData.Class.Name, "Function")) return true;
-            return false;
+            try
+            {
+                if (string.Equals(mirrorData.Class.Name, "Function")) return true;
+                return false;
+            }
+            catch (NullReferenceException)
+            {
+                return false;
+            }
+            catch (Exception)
+            {
+                return false;
+            }
```

Comment:
I don't think we need this. We can use the `IsFunction` property on the `MirrorData`class itself that tells us if it's a `Function `or not.

Reply:
Brilliant! Thanks for that.